File: Solutions/0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        for(auto str: strs) {
            string sss = str;
            sort(sss.begin(), sss.end());
            mp[sss].push_back(str);
        }
        vector<vector<string>> ans;
        for(auto pii: mp) {
            ans.push_back(pii.second);
        }
        return ans;
    }
};

File: Solutions/0049-group-anagrams/solution.ts
function groupAnagrams(strs: string[]): string[][] {
    const mp: Map<string, string[]> = new Map();
    for(const el of strs) {
        const sorted = el.split('').sort().join('');
        const arr = mp.get(sorted);
        if(arr) arr.push(el);
        else mp.set(sorted, [el]);
    }
    const ans: string[][] = [];
    for(const [key, val] of mp) {
        ans.push(val);
    }
    return ans;
};

